pkg/arvo/sys/zuse.hoon

```diff
@@ -6036,6 +6037,57 @@
       ?~  myn  ~
       ?:  =(~ t.myn)  (trip i.myn)
       (weld (trip i.myn) `tape`['/' $(myn t.myn)])
+    ::
+    ::  |base16: en/decode arbitrary MSB-first hex strings
+    ::
+    ++  base16
+      ~%  %base16  +  ~
+      |%
+      ++  en
+        ~/  %en
+        |=  a=(pair @ud @)  ::  $octs, but safe
```

Comment:
`$octs` is actually defined in multiple places, and referenced from each. I'll open a separate PR to clean that up.

Reply:
It's not immediately clear to me how that stops you from using `octs` here. There's no conflicting definition around, right?

Other than that, changes look good!

Reply:
I'm also not sure how @t is "unsafe" here tbh, like, does it not print bad bytes `\` escaped? That sounds like a printer bug, auras are always only advisory

Reply:
The specific crash that motivated this may be a bug, but it's at least a deliberate design bug (`+wood`, cord-escape, calls `+teff`, which includes a bunch of assertions). I could see arguing either way in the printer.

But this data structure exists solely for the purpose of representing octet-streams that cannot be losslessly encoded as atoms, so by definition, they cannot be cords. And I think it's fine for operations on atoms with auras that have a limited range to assume that their arguments are in that range. It would certainly be a pain to check such invariants anytime one handled such an atom. As we've discussed before, I'd like to see such validation performed more vigorously, but pushed to the edge of an input/type domain (by finishing `+sane` and enabling it in atom molds).

Reply:
Hm, am I wildly off base that the vast majority of `octs` constructed are in fact cords, whether or not they cannot be? Like, octs exists for representing octet streams we *cannot rule out* having trailing zeroes - I suppose my intuition here boils down to, if you `+curl` four textual urls and one `.gif`, the resulting type should both correctly represent the full range of the latter for `can` or fs sync or w/e, and print the common case as text and not as a decimal number.

Reply:
As far as I can tell, there's  currently two main use-cases for `$octs` or similar structures: http request/response bodies (in the latter case, possibly partial), and compatibility focused APIs like crypto functions (particularly hashes) and string-format en/decoding. It wouldn't surprise me if the majority of inputs in both cases are cords, but, even if so, the fact is incidental.

HTTP requests are now almost always routed to apps, where they're parsed/decoded and can be printed in an ad-hoc manner. Similarly, samples/products of various stdlib gates can be easily printed ad-hoc.

`+curl` is gone, but `+url` in still works in `:dojo`. That's the only use-case I can think of were we need to print arbitrary `$octs`. It'd be pretty easy to add a cast to cord there, gated by `(sane %t)`.

Reply:
Actually, that already happens -- see `+dy-print` in `:dojo`.